### PR TITLE
Documentation test branch

### PR DESCRIPTION
**Pull request by brynbarker**
Added a github action to run doxygen and update ibamr.github.io by pushing to the IBAMR-docs repository
This action also stores the doxygen log as an artifact

This does not fix any issue fully but is part of issue #1650 in that with this fix, any added tutorials will be correctly reflected on the github website. 

Next step will be to make it possible to build doxygen locally

### IBAMR Pull Request Checklist
- [ ] Does the test suite pass?
- [ ] Was clang-format run?
- [ ] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [ ] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [ ] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [ ] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [ ] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

File: .github/workflows/push-master.yml [hunk 1]
```diff
@@ -17,21 +17,15 @@ on:
   push:
     branches:
       - documentation-test-branch
```
Comment (drwells) [inline .github/workflows/push-master.yml:19]:
```suggestion
      - master
```

File: .github/workflows/push-master.yml [hunk 2]
```diff
@@ -17,21 +17,15 @@ on:
...
   deploy_documentation:
     runs-on: ubuntu-latest
-    name: Build IBAMR and run tests
+    name: Compile and deploy doxygen documentation
     env:
-      # TODO: this is a copy-and-paste of the Travis version. Once this is
-      # working we should clean it up.
-      TRAVIS_BUILD_NUMBER: ${{ github.run_number }}
-      TRAVIS_COMMIT: ${{ github.sha }}
+      TRAVIS_BUILD_DIR: ${{ github.workspace }}
```
Comment (drwells) [inline .github/workflows/push-master.yml:27]: Can we get rid of the rest of the travis variables at this point?

**Review by drwells — CHANGES_REQUESTED** [review]

**Comment by drwells** [comment]
Thanks a ton for this work. There's a lot more to do with the manual but getting things consistently building is the first step.

**Comment by drwells** [comment]
Can you also rebase and squash once everything is done?

**Comment by abarret** [comment]
Just FYI, the IBAMR doc page is empty as of this PR.

**Comment by drwells** [comment]
@brynbarker can you take a look? We need to get that on again ASAP.

File: .github/workflows/push-master.yml (hunk 2 above)
Comment (brynbarker) [reply .github/workflows/push-master.yml:27]: we need to keep this variable! removing it is what caused the ibamr doc page to be blank

**Comment by drwells** [comment]
Interesting - I don't know why we would need a travis variable when we don't use travis but lets sort that out later.

**Review by drwells — APPROVED** [review]

File: .github/workflows/push-master.yml (hunk 1 above)
Comment (drwells) [reply .github/workflows/push-master.yml:19]: We still need to make this change. Things are correctly rebuilt but we want it to be for `master`.

**Review by drwells — CHANGES_REQUESTED** [review]
We still need to change the branch name.

**Comment by drwells** [comment]
Nice. Would you mind squashing the commits so we have tidy history?

**Review by drwells — APPROVED** [review]
Let's deploy this and see what happens 😎

**Pull request merged**